Add sample_weight to the calculation of alphas in enet_path and LinearModelCV

s-banach opened the pull request:
#### Reference Issues/PRs
Fixes #22914.

#### What does this implement/fix? Explain your changes.
Modifies `_alpha_grid` function in `linear_model._coordinate_descent` to accept a `sample_weight` argument.
In addition to adding this argument, I have removed the argument `copy_X`, which no longer seems to be needed.

The function `_alpha_grid` is called in two places, `enet_path` and `LinearModelCV`.
The new `sample_weight` argument is not used by `enet_path`, but it is used by `LinearModelCV`.

#### Any other comments?
Thanks for your patience, this is my first PR and I'm trying my best.

s-banach committed:
Add sample_weight argument to _alpha_grid

Modifies _alpha_grid function in linear_model._coordinate_descent to accept a sample_weight argument.
In addition to adding this argument, I have removed the argument copy_X, which no longer seems to be needed.

The function _alpha_grid is called in two places, enet_path and LinearModelCV.
The new sample_weight argument is not used by enet_path, but it is used by LinearModelCV.

s-banach committed:
Revert formula for alpha_max

The old formula for alpha_max was correct, I misunderstood the way it handles Multitask regression.

s-banach committed:
Reshape matrix

Try to deal with Multitarget and single target regression

s-banach commented:
I wish you could hit undo when you submit an embarrassing one commit. Don't really understand this git thing.

jeremiedbb committed:
TST Fix test failing scipy nightly (#22935)

danifernandes-hub committed:
TST replace pytest.warns(None) in metrics/test_classification.py (#22888)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

thomasjpfan committed:
BLD Monkeypatch windows build to stablize build (#22693)

thomasjpfan committed:
FIX PowerTransformer Yeo-Johnson auto-tuning on significantly non-Gaussian data (#20653)

Co-authored-by: Olivier Grisel <olivier.grisel@gmail.com>

lorentzenchr commented:
@s-banach Thank you very much for giving *it* a try: solving this bug as well as opening your first PR. I'll try to review as time permits.

danifernandes-hub committed:
TST removed pytest.warns(None) in test_data.py (#22938)

danifernandes-hub committed:
TST replace pytest.warns(None) in test_function_transformer.py (#22937)

lorentzenchr committed:
MNT fix typo in tree test name (#22943)

lorentzenchr commented:
It would be a good idea to write a test in `sklearn/linear_model/test/test_coordinate_descent.py` where we check that the largest generated `alpha` gives all coefficients equal to zero. I couldn't find such a test, so we need to write a new one.

adam2392 committed:
[MAINT] Separate unit tests in `test_tree.py` for pickling and min_impurity_decrease (#22915)

thomasjpfan committed:
FIX Removes warning in HGBT when fitting on dataframes (#22908)

lorentzenchr committed:
ENH add sample_weight to sparse coordinade descent (#22808)

s-banach committed:
y.mean(axis=0)

Another case I didn't think about because I never use multi target regression.

varunagrawal committed:
FIX Fix recall in multilabel classification when true labels are all negative (#19085)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>
Co-authored-by: jeremiedbb <jeremiedbb@yahoo.fr>

s-banach committed:
Fixed product formula.

I don't need to make the sparse diagonal matrix here.

s-banach committed:
Change the formula again

I had this (1-mu) thing that I copied from somewhere but doesn't seem to make sense.

s-banach commented:
Dr. Lorentzen.
I do apologize for making so many commits.
My formula was originally wrong, because I copied it from some stack exchange post.
I believe it is correct now.

I can try writing a test. (Or I can leave that to you, if you no longer trust my code after seeing this big mess I've made!)
I think sometimes the coefficient will be exactly zero, and sometimes the value of alpha I compute results in a coefficient on the order of 1e-16. Is it important that it be exactly zero, or just approximately zero?

s-banach committed:
Else product may be matrix

If product is left as a scipy matrix, then product**2 will not be "element-wise power"

lorentzenchr commented:
@s-banach No problem at all. In the end, you need to write such a test because without it, we can't merge this PR. It is good practice (e.g. *test driven development*) to write such a test early. Tests are there to **proof** that the implementation does exactly what it is supposed to do.

What I notice is that you changed to code quite a lot and I fear that this way, it might be hard to detect errors. I would suggest to make smaller changes, i.e. only introduce the additional functionality - and a test for it - nothing more.

GaelVaroquaux committed:
DOC Update comms team (#22942)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

s-banach committed:
Update _coordinate_descent.py

Now the logic should be unchanged from original, if sample_weight is None

s-banach commented:
@lorentzenchr
In the latest commit, I have followed your suggestion.
The code should be identical to the old code, except in the case `(sample_weight is not None) and fit_intercept`.
I will think about a test.

jjerphan committed:
TST use global_dtype in feature_selection/tests/test_mutual_info.py (#22677)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Jérémie du Boisberranger <jeremiedbb@users.noreply.github.com>

thomasjpfan committed:
DOC Makes Sphinx reference to Bunch a class (#22948)

JosephSM committed:
DOC fix typo (#22958)

danifernandes-hub committed:
TST replace pytest.warns(None) in preprocessing/test_common.py (#22936)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

danifernandes-hub committed:
TST replace pytest.warns(None) in metrics/tests/test_pairwise.py (#22957)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

TheisFerre committed:
TST Replace pytest.warns(None) in test_gpr.py (#22959)

Co-authored-by: Theis Hjortkjær <theis@Theiss-MacBook-Pro.local>

danifernandes-hub committed:
TST replace pytest.warns(None) in metrics/cluster/test_unsupervised.py (#22956)

danifernandes-hub committed:
TST replace pytest.warns(None) in metrics/cluster/test_supervised.py (#22955)

TheisFerre committed:
TST Replace pytest.warns(None) in test_gpc.py (#22960)

danifernandes-hub committed:
TST Replace pytest.warns(None) in utils/tests/test_utils.py (#22961)

yiyangq committed:
TST Replace pytest.warns(None) in manifold/tests/test_t_sne.py (#22963)

yiyangq committed:
TST Replace pytest.warns(None) in feature_extraction/tests/test_text.py (#22962)

glemaitre committed:
FIX make coef_ in PLS estimator consistent with linear models (#22016)

Co-authored-by: jeremie du boisberranger <jeremiedbb@yahoo.fr>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

JihaneBennis committed:
ENH Preserving dtypes for ICA (#22806)



Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>
Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

lorentzenchr committed:
ENH migrate GLMs / TweedieRegressor to linear loss (#22548)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

thomasjpfan committed:
MNT Update to black 22.3.0 to resolve click error (#22983)

* MNT Update to black 22.3.0 to resolve click error

* STY Update for new black version

thomasjpfan committed:
FEA Add DecisionBoundaryDisplay (#16061)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Loïc Estève <loic.esteve@ymail.com>

jsilke committed:
MNT accelerate examples/kernel_approximation/plot_scalable_poly_kernels.py (#22903)

Co-authored-by: sveneschlbeck <sven.eschlbeck@t-online.de>
Co-authored-by: Sven Eschlbeck <66701689+sveneschlbeck@users.noreply.github.com>
Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

thomasjpfan committed:
MAINT Refactor vector sentinel into utils (#22728)

thomasjpfan committed:
DOC Fixes nav bar by dynamically changing searchbar size (#22954)

thomasjpfan committed:
MNT Uses memoryviews in tree criterion (#22921)

Co-authored-by: Julien Jerphanion <git@jjerphan.xyz>

jnothman committed:
DOC no longer funded by sydney university (#22980)

lorentzenchr committed:
ENH enable LSQR solver with intercept term in Ridge with sparse input (#22950)

* ENH support Ridge(fit_intercept=True, solver="lsqr") for sparse input

* DOC add whatsnew

* FIX set solver to lsqr

* DOC _get_rescaled_operator

* DOC add more details

thomasjpfan committed:
DOC Link directly developer docs in the navbar (#22550)

jeremiedbb committed:
[MRG] Refactor MiniBatchDictionaryLearning and add stopping criterion (#18975)


Co-authored-by: Olivier Grisel <olivier.grisel@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

Commit:
DOC: fix typo (#22994)

jjerphan committed:
TST use global_dtype in sklearn/neighbors/tests/test_neighbors.py (#22663)

Co-authored-by: Jérémie du Boisberranger
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

2357juan committed:
DOC Update notebook style for plot_bayesian_ridge_curvefit (#22916)

Co-authored-by: Juan Gomez <jgomez75@gatech.edu>

Ben3940 committed:
DOC fix docstring of EllipticEnvelope.fit (#22997)

Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

adrinjalali committed:
MNT Fix pytest random seed pluging with vscode test discovery (#22976)

Co-authored-by: Thomas J. Fan <thomasjpfan@gmail.com>

lorentzenchr committed:
TST tight and clean tests for Ridge (#22910)

* MNT replace pinvh by solve

* DOC more info for svd solver

* TST rewrite test_ridge

* MNT remove test_ridge_singular

* MNT restructure into several tests

* MNT remove test_toy_ridge_object

* MNT remove test_ridge_sparse_svd

This is tested in test_ridge_fit_intercept_sparse_error.

* TST exclude cholesky from singular problem

* CLN two fixes

* MNT parametrize test_ridge_sample_weights

* MNT restructure test_ridge_sample_weights

* CLN tighten tolerance for sag solver

* CLN try to fix saga tolerance

* CLN make test_ridge_sample_weights nicer

* MNT remove test_ridge_regression_sample_weights

* MNT rename to test_ridge_regression_sample_weights

* CLN make test_ridge_regression_unpenalized pass for all random seeds

* CLN make tests pass for all random seeds

* DOC fix typos

* TST skip cholesky for singular problems

* MNT move up test_ridge_regression_sample_weights

* CLN set skip reason as comment

inclusive-coding-bot committed:
DOC Switch to gender neutral terms for sister function (#23003)

randomgeek78 committed:
FIX ColumnTransformer.get_feature_names_out with string slices (#22913)

* FIX ColumnTransformer.get_feature_names_out with string slices

* Fixed black formatting

* Added tests

* Updated whats new 1.1 section

* Simplified logic and added slices containing two elements

* Fixed black linting

* Linting cleanup, revert API change

* Fixed linting

jeremiedbb committed:
Rename triage team to contributor experience team (#22970)

aj-white committed:
DOC Ensures that homogeneity_score passes numpydoc validation (#23006)

jeremiedbb committed:
CI use circleci artifact redirector GH action (#22991)

jeremiedbb committed:
MNT remove artifact_path file now unused (#23012)

jjerphan committed:
MAINT Convert OpenMP scheduling to 'static' in pairwise distances radius neighborhood (#22829)

lorentzenchr committed:
DOC precise stopping criteria for coordinate descent

lorentzenchr committed:
MNT Revert "DOC precise stopping criteria for coordinate descent"

This reverts commit cc3fdbfe3f3c661a83e82fc0263c1cf28f821a5b.

jjerphan committed:
TST use global_dtype in sklearn/cluster/tests/test_optics.py (#22668)

* TST Adapt test_optics.py to test implementations on 32bit datasets

* TST Use global_dtype

* Apply review comments

jjerphan committed:
TST use global_dtype in sklearn/manifold/tests/test_locally_linear.py (#22676)

aj-white committed:
DOC Ensure completeness_score passes numpydoc validation (#23016)

PierreAttard committed:
MNT ensure creation of dataset is deterministic in SGD  (#19716)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>
Co-authored-by: Jérémie du Boisberranger <34657725+jeremiedbb@users.noreply.github.com>

s-banach committed:
catch up with scikit-learn main branch

s-banach committed:
Merge branch 'main' of https://github.com/s-banach/scikit-learn

s-banach committed:
Simple test for alpha_max with sample_weight

It merely asserts that alpha_max is large enough to force the coefficients to zero. It doesn't test anything else about alpha_max or _alpha_grid.

s-banach commented:
Closing because I copy-pasted from your repo into my repo and now it says I'm making 66 commits.

s-banach commented:
Thanks, I should learn more about using git.
Since I didn't know what to do, I opened a new PR https://github.com/scikit-learn/scikit-learn/pull/23045.
The code is simpler now, since you've added sample_weight to _preprocess_data.
